### DISCUSSION TITLE
Npm Run Watch issue with JIT using Laravel Mix with Tailwind

### DISCUSSION THREAD

**Discussion by chrisgrim** [tailwindlabs/tailwindcss#7589] [Help] [answered]
"laravel-mix": "^6.0.6",
"laravel-mix-tailwind": "^0.1.2",
"postcss": "^8.4.6",
"postcss-import": "^12.0.1",
"tailwindcss": "^3.0.21",

I recently discovered tailwind and I can't explain how much I love it. I set it up for one of my WordPress projects and the JIT compiler was amazing. I recently have tried to set it up with Laravel mix and I am not having much luck.
When I run npm run watch it works, however if I save a Vue file in my project it looks like it recompiles but if I created a custom class like h-[43rem] it won't be there. I have to cancel npm run watch and then run it again. This does work but it gets tedious to have to cancel the npm run watch every time I add a new class. 

I am using Laravel mix tailwind and in my webpack.mix.js file I have 
```
const mix = require('laravel-mix');

require('laravel-mix-tailwind');

mix.sass('resources/sass/app-create.scss', 'public/assets')
    .sass('resources/sass/app-admin.scss', 'public/assets')
    .sass('resources/sass/app-lite.scss', 'public/assets')
    .postCss('resources/css/app.css', 'public/assets', [
        require('tailwindcss'),
    ])
    .js('resources/js/app.js', 'public/assets')
    .extract(['vue'])
    .webpackConfig({
        output: {
            chunkFilename: '[name].js?id=[chunkhash]',
        }
    })
    .tailwind()
    .vue();
```
In my tailwind.js file (this is what laravel mix tailwind uses instead of the tailwind.config.js)
```
module.exports = {
    mode: 'jit',
    content: [
        "./resources/**/*.blade.php",
        "./resources/**/*.js",
        "./resources/**/*.vue",
    ],
    theme: {
        fontSize: {
            'xs': '.75rem',
            'sm': '.875rem',
            'tiny': '.875rem',
            'base': '1rem',
            'lg': '1.125rem',
            'xl': '1.25rem',
            '1xl': '1.35rem',
            '2xl': '1.5rem',
            '3xl': '1.875rem',
            '4xl': '2.25rem',
            '5xl': '3rem',
            '6xl': '4rem',
            '7xl': '5rem',
        },
        extend: {
            spacing: {
                '128': '32rem',
            }
        }
    },
    plugins: [],
}
```

and in my package.json I have
```
    "scripts": {
        "dev": "npm run development",
        "development": "TAILWIND_MODE=build NODE_ENV=development mix",
        "watch": "TAILWIND_MODE=watch NODE_ENV=development mix watch",
        "watch-poll": "mix watch -- --watch-options-poll=1000",
        "hot": "mix watch --hot",
        "prod": "npm run production",
        "production": "mix --production"
    },
```
Thanks so much!!

**Comment by adamwathan** [comment]
Hey! Can you please provide a full reproduction? Something we can actually clone and see the issue happening would be much easier to work with than trying to recreate your whole setup from some code snippets 👍 Especially because there is likely more to the story than is actually shared here, and the real problem might be in a different file or to do with the structure of the project or something else.

**Reply by chrisgrim** [reply]
Ahh ok Thanks!

**Reply by chrisgrim** [reply]
Ok Ill put it together. However just to be clear this isn't how JIT is supposed to behave? It should work with laravel mix?

**Comment by chrisgrim** [comment, marked as answer]
I spent this morning making a new project and then figuring out what was going wrong. It turns out I didn't have the latest Laravel mix! I updated to 6.0.43 and it worked. Thanks so much!
